Cannot parse disabled const in impl

The following snippet is rejected by GCCRS even though the code is disabled and should be accepted.

I tried this code:

```rust
fn main() {}

#[cfg(FALSE)]
impl X {
    const Y: u8;
}
```

I expected to see this happen: Compiles correctly

Instead, this happened:

```
./test.rs:5:16: error: expecting ‘=’ but ‘;’ found
    5 |     const Y: u8;
      |                ^
./test.rs:6:1: error: failed to parse inherent impl item in inherent impl
    6 | }
      | ^
./test.rs:6:1: error: failed to parse item in crate
```

### Meta

- c7728c8020334fecb47789163ba94326aab6208c

--- a/gcc/rust/ast/rust-item.h
+++ b/gcc/rust/ast/rust-item.h
@@ -2618,6 +2618,13 @@ public:
       const_expr (std::move (const_expr)), locus (locus)
   {}
 
+  ConstantItem (std::string ident, Visibility vis, std::unique_ptr<Type> type,
+		std::vector<Attribute> outer_attrs, location_t locus)
+    : VisItem (std::move (vis), std::move (outer_attrs)),
+      identifier (std::move (ident)), type (std::move (type)),
+      const_expr (nullptr), locus (locus)
+  {}
+
   ConstantItem (ConstantItem const &other)
     : VisItem (other), identifier (other.identifier), locus (other.locus)
   {

--- a/gcc/rust/parse/rust-parse-impl.h
+++ b/gcc/rust/parse/rust-parse-impl.h
@@ -4783,6 +4783,16 @@ Parser<ManagedTokenSource>::parse_const_item (AST::Visibility vis,
   // parse constant type (required)
   std::unique_ptr<AST::Type> type = parse_type ();
 
+  // A const with no given expression value
+  if (lexer.peek_token ()->get_id () == SEMICOLON)
+    {
+      lexer.skip_token ();
+      return std::unique_ptr<AST::ConstantItem> (
+	new AST::ConstantItem (std::move (ident), std::move (vis),
+			       std::move (type), std::move (outer_attrs),
+			       locus));
+    }
+
   if (!skip_token (EQUAL))
     {
       skip_after_semicolon ();

--- a/gcc/testsuite/rust/compile/issue-2665.rs
+++ b/gcc/testsuite/rust/compile/issue-2665.rs
@@ -1,0 +1,6 @@
+fn main() {}
+
+#[cfg(FALSE)]
+impl X {
+    const Y: u8;
+}

